balancer/rls/config.go

```diff
@@ -195,13 +195,13 @@ func parseRLSProto(rlsProto *rlspb.RouteLookupConfig) (*lbConfig, error) {
 	parsedTarget, err := url.Parse(lookupService)
 	if err != nil {
 		// url.Parse() fails if scheme is missing. Retry with default scheme.
-		parsedTarget, err = url.Parse(resolver.GetDefaultScheme() + ":///" + lookupService)
+		parsedTarget, err = url.Parse(resolver.GetDefaultSchemeOrPassthrough() + ":///" + lookupService)
```

Comment:
I don't really love these names, am open to changing them. Maybe something like `GetOverridenSchemeOrPassthrough`?

Reply:
Hmm, can we avoid changing the existing things entirely, please, and keep the behavior the same?

I think that could work if we have a way to determine whether `SetDefaultScheme` was ever called by the user (ideally only exposed via `internal`), and a way to distinguish whether the user was directly calling `NewClient` or `Dial`.